src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
@luisgabriel could do the problem actually came into play because I improved the definition for `tryCatch` :smile: 

`export function tryCatch<T extends Function>(fn: T): T`, so it knows the type of the function it will flow through to the invocation, which gives you type safety there instead of just pretending it's a function.

I'll add in the errorObject type change, it is probably small enough that it shouldn't be a big deal.